hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -306,8 +307,13 @@ def _convert_item(cls, item: t.Dict[str, t.Any]):
 # to make them easier to spot in the wild
 
 
-def update_config(config: HMAConfig) -> None:
-    """Update or create a config. No locking or versioning!"""
+def update_config(
```

Comment:
The insert_only vs upsert difference, IMO, is significant.

How do you feel about two separate methods for this? 
`config.upsert_config` and `config.insert_config`. Explicit, top-level and hard to get wrong.

Reply:
> Would have preferred the config.py changes as a separate PR. It is sizeable and more-importantly, reviewable separately from the rest of the PR.
> 
> I have various inline recommendations.
> 
> While right now, the code works, from a better-engineering perspective, there are quite a few misses which I would want to catch early. Let me know if I can help with implementing any of the recommendations.
> 
> I am also fine to punt on this for now and revisit when we have time, but I'd strongly urge fixing these issues now.

yeah thanks for that!
and I absolutely agree to make changes now to avoid future work.
I will make those changes and separate the PR of config.py changes